### PR TITLE
feat(matchDetail): upgrade to new API and looks

### PR DESCRIPTION
**Pull request by LarsSelbekk**
- bug(api): fix infinite recursion on api error
- feat(matchDetail): upgrade to new API and looks

### CODE REVIEW

**Comment by LarsSelbekk** [comment]
Web capture_17-6-2023_33423_localhost
Web capture_17-6-2023_22820_localhost
Web capture_17-6-2023_22659_localhost
Web capture_17-6-2023_2268_localhost
Web capture_17-6-2023_22541_localhost


Early centered version, I prefer left-aligned & it's more consistent with the rest of the site:
Web capture_17-6-2023_12712_localhost

**Comment by LarsSelbekk** [comment]
[Oppdatere Match Detail til å bruke nyeste versjon av APIet](https://trello.com/c/4HgSwdlN/668-oppdatere-match-detail-til-%C3%A5-bruke-nyeste-versjon-av-apiet)

**Comment by LarsSelbekk** [comment]
[Oppdatere Match Detail til å bruke nyeste versjon av APIet](https://trello.com/c/4HgSwdlN/668-oppdatere-match-detail-til-%C3%A5-bruke-nyeste-versjon-av-apiet)

**Review by AdrianAndersen — APPROVED** [review]
Had a very breif read through. Not spotting any obvious problems

File: src/components/matches/matches-helper.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      {children}
+    </Typography>
+  );
+};
+
+export function calculateItemStatuses<T extends MatchWithDetails>(
+  match: T,
+  // surpressing because it thinks "match" is an actual variable
+  // eslint-disable-next-line no-unused-vars
+  expectedItemsSelector: (match: T) => string[],
```
[the remark this answers is not in the document]
Comment (LarsSelbekk) [reply src/components/matches/matches-helper.tsx:23]: No I like it, it's just bugged in this particular case

File: src/api/api.ts [hunk 2]
```diff
@@ -16,13 +15,21 @@ export const get = async <T = any>(url: string, query?: string) => {
...
-      return fetchTokensAndGet(url);
+      if (
+        (error.status === 401 || error.response.status === 401) &&
+        !noRetryTokens
+      ) {
+        await fetchNewTokens();
+        return await get(url, query, true);
+      }
+
+      throw new Error("Unknown API error");
```
[the remark this answers is not in the document]
Comment (LarsSelbekk) [reply src/api/api.ts:32]: Am I supposed to do something about that or can we resolve this comment?

File: src/components/matches/MatchDetail.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  if (matchesError instanceof NotFoundError) {
+    return (
+      <Alert severity="error">
+        Kunne ikke finne en overlevering med ID {matchId}.
+      </Alert>
+    );
+  }
+
+  if (tokenError || matchesError) {
+    return <Alert severity="error">En feil har oppstått.</Alert>;
```
[the remark this answers is not in the document]
Comment (LarsSelbekk) [reply src/components/matches/MatchDetail.tsx:50]: Done

File: src/components/matches/matches-helper.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (LarsSelbekk) [reply src/components/matches/matches-helper.tsx:23]: Can I resolve?

File: src/components/matches/MeetingInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      </Typography>
+      <Box sx={{ display: "flex", marginTop: ".2rem", alignItems: "center" }}>
+        <ScheduleIcon sx={{ marginRight: ".2rem" }} />
+        {(meetingTime && (
+          <>
+            <Typography fontWeight="bold" variant={"subtitle1"}>
+              {formatDatetime(new Date(meetingTime))}
+            </Typography>
+          </>
+        )) || <>Du kan møte opp når som helst i løpet av dagen</>}
```
[the remark this answers is not in the document]
Comment (LarsSelbekk) [reply src/components/matches/MeetingInfo.tsx:26]: Done

**Pull request merged**